Tests/TestingApps/PaywallsTester/PaywallsTester/Views/AppContentView.swift

```diff
@@ -139,16 +134,75 @@ struct AppContentView: View {
         }
     }
 
-    private var isPurchasesConfigured: Bool {
-        return self.customerInfoStream != nil
+    private func reconfigure(for mode: Configuration.Mode) {
+        configuration.reconfigure(for: mode)
+        self.observeCustomerInfoStream()
     }
 
+    private func observeCustomerInfoStream() {
+        self.customerInfoTask?.cancel()
+        self.customerInfoTask = Task {
+            if Purchases.isConfigured {
+                for await info in Purchases.shared.customerInfoStream {
+                    self.customerInfo = info
+                    self.showingDefaultPaywall = self.showingDefaultPaywall && info.activeSubscriptions.count == 0
+                }
+            }
+        }
+    }
+
+    private func descriptionForCurrentMode() -> String {
+
+        switch self.configuration.currentMode {
+        case .custom:
+            return "the API set locally in Configuration.swift"
+        case .testing:
+            return "the Paywalls Tester app in RevenueCat Dashboard"
+        case .demos:
+            return "Demos"
+        }
+
+    }
+
+}
+private struct ProminentButton: View {
+    var title: String
+    var action: () -> Void
+    var background: Color = .accentColor
+
+    var body: some View {
+        Button(action: action) {
+            Text(title)
+                .bold()
+                .padding()
+                .frame(maxWidth: .infinity)
+                .background(background)
+                .foregroundColor(.white)
+                .cornerRadius(10)
+        }
+    }
+}
+
+private struct ConfigurationButton: View {
+    var title: String
+    var mode: Configuration.Mode
+    @ObservedObject var configuration: Configuration
+    var action: () -> Void
+
+    var body: some View {
+        ProminentButton(
+            title: title,
+            action: action,
+            background: configuration.currentMode == mode ? Color.gray : Color.accentColor
+        )
+        .disabled(configuration.currentMode == mode)
```

Comment:
Explicit self here too:
```suggestion
            title: self.title,
            action: self.action,
            background: self.configuration.currentMode == self.mode ? Color.gray : Color.accentColor
        )
        .disabled(self.configuration.currentMode == self.mode)
```